src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -1821,108 +1809,64 @@ private static ITypeSymbol GetMemberType(ISymbol memberSymbol)
-                types = SpecializedCollections.EmptyEnumerable<TypeInferenceInfo>();
-
                 // If we are position based, then we have to be after the return statement.
                 if (previousToken.HasValue && previousToken.Value != returnStatement.ReturnKeyword)
                 {
-                    return;
+                    return SpecializedCollections.EmptyEnumerable<TypeInferenceInfo>();
                 }
 
-                var ancestor = returnStatement.AncestorsAndSelf().FirstOrDefault(e => e is AnonymousFunctionExpressionSyntax || e is LocalFunctionStatementSyntax);
+                var ancestor = returnStatement.FirstAncestorOrSelf<SyntaxNode>(n => n.IsReturnableConstruct());
 
-                if (ancestor is LambdaExpressionSyntax lambdaExpression)
-                {
-                    // If we're in a lambda, then use the return type of the lambda to figure out what to
-                    // infer.  i.e.   Func<int,string> f = i => { return Goo(); }
-                    types = InferTypeInLambdaExpression(lambdaExpression);
-                    isAsync = lambdaExpression.AsyncKeyword.Kind() != SyntaxKind.None;
-                    return;
-                }
-                else if (ancestor is AnonymousMethodExpressionSyntax delegateExpression)
-                {
-                    // If we are inside a delegate then use the return type of the Invoke Method of the delegate type
-                    var delegateType = InferTypes(delegateExpression).FirstOrDefault().InferredType;
-                    if (delegateType != null && delegateType.IsDelegateType())
-                    {
-                        var delegateInvokeMethod = delegateType.GetDelegateType(this.Compilation).DelegateInvokeMethod;
-                        if (delegateInvokeMethod != null)
-                        {
-                            types = SpecializedCollections.SingletonEnumerable(new TypeInferenceInfo(delegateInvokeMethod.ReturnType));
-                            isAsync = delegateExpression.AsyncKeyword.Kind() != SyntaxKind.None;
-                            return;
-                        }
-                    }
-                }
-                else if (ancestor is LocalFunctionStatementSyntax localFunctionStatement)
-                {
-                    // If we are inside a local function then use the return type of the local function
-                    var methodSymbol = (IMethodSymbol)SemanticModel.GetDeclaredSymbol(localFunctionStatement);
-                    types = SpecializedCollections.SingletonEnumerable(new TypeInferenceInfo(methodSymbol.ReturnType));
-                    isAsync = methodSymbol.IsAsync;
-                    return;
-                }
+                return ancestor is AnonymousFunctionExpressionSyntax anonymousFunction
+                    ? InferTypeInAnonymousFunctionExpression(anonymousFunction)
+                    : InferTypeInDeclaration(ancestor);
```

Comment:
if ancestor is null, is this ok?  does InferTypeInDeclaration handle that?

Reply:
I don't think it does. It eventually gets passed to `SemanticModel.GetDeclaredSymbol`.

Reply:
When could this occur? In a scripting scenario? Or what would be the best way to test?

Reply:
not certain.  but we shoudl be resilient to it.  you could try a scripting scenario.

Reply:
➡️ Thanks, fixed in 7003dfecf612d69a7f3e474c63e8bab97803b6bb for both `return` and `yield return` (but I couldn't reproduce this crash using any feature)